docs/PORTABLE_SERVICES.md

```diff
@@ -345,15 +345,28 @@ was loaded from. In case extensions are used, additionally there will be a
 (i.e.: `RootImage=` or `RootDirectory=`), and one `PORTABLE_EXTENSION=` field per
 each extension image used.
 
+The os-release file from the portable image will be parsed parsed and the `ID`,
+`VERSION_ID`, `IMAGE_VERSION`, `IMAGE_ID`, `BUILD_ID` fields are added as
+structured metadata to the journal log entries. In case extensions are used, the
+same fields but prefixed by `SYSEXT_` are parsed from each extension-release file,
+and are appended to the journal log entries, but the values are prefixed with the
+extension name, so that when multiple extensions are used at the same time it is
+still possible to associate a field to an image.
+
 For example, a portable service `app0` using two extensions `app0.raw` and
-`app1.raw`, and a base layer `base.raw`, will create log entries with the
-following fields:
+`app1.raw` (with `SYSEXT_VERSION_ID=` `0` and `1` in their respective
+extension-releases), and a base layer `base.raw` (with `VERSION_ID=10` and
+`ID=debian` in os-release), will create log entries with the following fields:
 
 ```
 PORTABLE=app0.raw
 PORTABLE_ROOT=base.raw
 PORTABLE_EXTENSION=app0.raw
 PORTABLE_EXTENSION=app1.raw
+VERSION_ID=10
+ID=debian
```

Comment:
hmm, so we end up adding this into all log messages written by services. we probably should be a bit careful with namespace conflicts there. i.e. not use super duper generic field names such as "ID=" for this, because that sounds like something apps might want to use for themselves. This is in particular relevant since this ID field here is an OS ID, but that is not clear at all if you just look at the field name.

hence, maybe prefix these fields with OS_RELEASE_ or so? That would make them a lot more specific and clashes much less likely. Moreover, it would leave the door open that we eventually attach these fields entirely generically and automatically also for non-portable services.

Reply:
yes I was thinking the same, and also the format is a bit unwieldly, and we get a lot of fields added, so instead I'll add a single field per image, called PORTABLE_ID and compose the value from an ID and a VERSION_ID, depending on what's available